apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -457,26 +461,84 @@ type EmbeddedObjectMeta struct {
 	Name string `json:"name"`
 
 	// +optional
-	Labels map[string]string `json:"labels,omitempty"`
+	Metadata *CassandraDatacenterMeta `json:"dcMetadata,omitempty"`
+}
 
+type CassandraDatacenterMeta struct {
+	// labels/annotations for the CassandraDatacenter object
 	// +optional
-	Annotations map[string]string `json:"annotations,omitempty"`
-}
+	Resource *meta.MetaTags `json:"resource,omitempty"`
 
-func (in *EmbeddedObjectMeta) GetAnnotations() map[string]string {
-	return in.Annotations
-}
+	// labels/annotations that will be applied to all resources
+	// created by the CassandraDatacenter
+	// +optional
+	CommonLabels map[string]string `json:"commonLabels,omitempty"`
+
+	// labels/annotations for the pod components
+	// +optional
+	Pods *meta.MetaTags `json:"pods,omitempty"`
 
-func (in *EmbeddedObjectMeta) SetAnnotations(annotations map[string]string) {
-	in.Annotations = annotations
+	// labels/annotations for all of the CassandraDatacenter service components
+	ServiceConfig *cassdcapi.ServiceConfig `json:"services,omitempty"`
```

Comment:
So, using `ServiceConfig` directly is easy, but it exposes fields like `additionalLabels` and `additionalAnnotations`. It would be nicer to reuse `MetaTags`. I think we can achieve that with something like this:

```go

// CassandraDatacenterServicesMeta is very similar to cassdcapi.ServiceConfig and is passed to
// cass-operator in the AdditionalServiceConfig field of the CassandraDatacenter spec.
// +kubebuilder:object:generate=true
type CassandraDatacenterServicesMeta struct {
	DatacenterService     MetaTags `json:"dcService,omitempty"`
	SeedService           MetaTags `json:"seedService,omitempty"`
	AllPodsService        MetaTags `json:"allPodsService,omitempty"`
	AdditionalSeedService MetaTags `json:"additionalSeedService,omitempty"`
	NodePortService       MetaTags `json:"nodePortService,omitempty"`
}

func (in CassandraDatacenterServicesMeta) ToCassAdditionalServiceConfig() cassdcapi.ServiceConfig {
	return cassdcapi.ServiceConfig{
		DatacenterService: cassdcapi.ServiceConfigAdditions{
			Annotations: in.DatacenterService.Annotations,
			Labels:      in.DatacenterService.Labels,
		},
		SeedService: cassdcapi.ServiceConfigAdditions{
			Annotations: in.SeedService.Annotations,
			Labels:      in.SeedService.Labels,
		},
		AdditionalSeedService: cassdcapi.ServiceConfigAdditions{
			Annotations: in.AdditionalSeedService.Annotations,
			Labels:      in.AdditionalSeedService.Labels,
		},
		AllPodsService: cassdcapi.ServiceConfigAdditions{
			Annotations: in.AllPodsService.Annotations,
			Labels:      in.AllPodsService.Labels,
		},
		NodePortService: cassdcapi.ServiceConfigAdditions{
			Annotations: in.NodePortService.Annotations,
			Labels:      in.NodePortService.Labels,
		},
	}
}
```

Wdyt?